app/javascript/components/workflows/workflow_payload.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Loading, Tabs, Tab,
+} from 'carbon-components-react';
+import AWSSfnGraph from '@tshepomgaga/aws-sfn-graph';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+import '@tshepomgaga/aws-sfn-graph/index.css';
+
+const WorkflowPayload = ({ recordId }) => {
+  const tabLabels = [
+    { name: 'text', text: __('Text') },
+    { name: 'graph', text: __('Graph') },
+  ];
+  const [data, setData] = useState({
+    isLoading: true,
+    payload: '',
+    activeTab: tabLabels[0].name,
+  });
+
+  useEffect(() => {
+    API.get(`/api/configuration_script_payloads/${recordId}?expand=resources`)
+      .then((response) => {
+        setData({
+          isLoading: false,
+          payload: response.payload,
+        });
+      });
+  }, [recordId]);
+
+  if (data.isLoading) {
+    return (
+      <div className="loadingSpinner">
+        <Loading active small withOverlay={false} className="loading" />
+      </div>
+    );
+  }
+
+  /** Function to set the active tab name */
+  const onTabSelect = (name) => {
+    setData({
+      ...data,
+      activeTab: name,
+    });
+  };
+
+  /** Function to render the code mirror component */
+  const error = false;
+  const renderCodeSnippet = () => (
+    <CodeMirror
+      className={`miq-codemirror workflow-payload ${error ? 'has-error' : ''}`}
+      options={{
+        mode: 'application/ld+json',
```

Comment:
As discussed this should be
```suggestion
        mode: { name: "javascript", json: true }
```

Or perhaps just use the CodeEditor component we already have.

---

We might want to take the time to check the payload_type also and then this would be a function to turn the payload_type into the mode value.

Reply:
I tried to integrate the CodeEditor here and it throws an error like this
<img width="635" alt="image" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/87487049/97616aaa-b322-4391-9f35-2e983b34938c">

The `CodeEditor` needs to be inside a form since it contains `useField` to assign values to the DDF form.

So will have to refactor the code here to use this.